### PR TITLE
Feature/dynamixel position

### PR DESCRIPTION
**Pull request by ZacharyTaylor**
Adds a driver to allow the px4 to interface with dynamixel servos that use the dynamixel 2.0 protocol (http://support.robotis.com/en/product/actuator/dynamixel_pro/communication.htm)

This driver uses a very small subset of the possible dynamixel options, only implementing a broadcast write instruction and broadcast read instruction (currently unused).

It allows using the dynamixels in extended position control mode taking commanded angles on the actuator control group 2. It disables all feedback from the servos as many dynamixel servos use a 1 wire interface and waiting for the response significantly limits the bandwidth.

I have not included it in any builds, as I figure this would be quite a rarely used driver that only a small number of people would be interested in.

Tested with 6 dynamixel XL430-W250-T servos.

### CODE REVIEW

File: src/drivers/dynamixel/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+############################################################################
+#
+#   Copyright (c) 2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/dynamixel/CMakeLists.txt]: copyright years

File: src/drivers/dynamixel/message.c [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+ *
+ ****************************************************************************/
+
+#include "message.h"
+#include <math.h>
+
+uint16_t calc_crc(uint8_t *data_blk_ptr, uint16_t data_blk_size)
+{
+	uint16_t i, j;
+	uint16_t crc_table[256] = {
```
Comment (dagar) [inline src/drivers/dynamixel/message.c]: Which CRC is this? There are a few others already in the system. I'll find them.

**Comment by dagar** [comment]
How did you want this to work with the mixers? I see it's sitting on actuator controls 2, but how do you map it to an index?

**Comment by dagar** [comment]
Which board do you typically use? This could be included in the main target configs (px4fmu-v3, v4, v4pro, v5) to make sure it actually builds and doesn't break without anyone noticing.

**Comment by ZacharyTaylor** [comment]
I have been using this driver code on a px4fmu-v2 on an omnidirectional hexacopter. The system doesn't use a standard mixer so I forgot to consider how it could fit into the normal control pipeline. 

Actually I wasn't too sure about using the actuator control message at all, as it seems designed for a slightly different role and usually takes normalized values, but it seemed cleaner than creating a custom dynamixel message. I could try modify it so that it fits closer to how the standard actuator outputs work if you feel this would be of benefit.

**Comment by dagar** [comment]
Even if it doesn't fit into the system normally at the moment I think it's still fine as an example.

**Comment by ZacharyTaylor** [comment]
I am going to be out of the office the next two weeks but I agree with the comments and will make the changes once I get back

File: src/drivers/dynamixel/message.c (hunk 2 above)
Comment (ZacharyTaylor) [reply src/drivers/dynamixel/message.c]: the CRC table is for the CRC-16. The poly is 0x8005 (x^16 + x^15 + x^2 + 1). I had a look through the px4 code and the only matching one I found was here https://github.com/PX4/Firmware/blob/master/msg/templates/urtps/microRTPS_transport.cpp#L45
However, the crc check is inside a protected part of the class, and so not too easy to just include and use.

**Comment by TSC21** [comment]
@ZacharyTaylor can you please finish addressing the review comments and rebase? Thanks

**Pull request closed without merging**